testcases/OpTestHMIHandling.py

```diff
@@ -221,28 +138,26 @@ def clearGardEntries(self):
     #                          BMC_CONST.HMI_PROC_RECV_ERROR_MASKED: proc_recv_error_masked
     #                          BMC_CONST.HMI_MALFUNCTION_ALERT: malfunction_alert
     #                          BMC_CONST.HMI_HYPERVISOR_RESOURCE_ERROR: hypervisor resource error
-    #
-    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
-    #
-    def testHMIHandling(self, i_test):
+    def _testHMIHandling(self, i_test):
         l_test = i_test
-        self.test_init()
+        self.init_test()
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+
         l_con = self.cv_SYSTEM.sys_get_ipmi_console()
         self.cv_IPMI.ipmi_host_login(l_con)
-        self.cv_IPMI.ipmi_host_set_unique_prompt(l_con)
```

Comment:
Here ipmi console process is broken. Can you change it to below. By changing like this make me test case working
        l_con = self.cv_SYSTEM.sys_get_ipmi_console()
        c = l_con.get_console()
        self.cv_IPMI.ipmi_host_login(c)


Can you document these API's in OpTestSystem so it will be easy for anyone to follow this protocol before writing any ipmi based testcase.
console = self.cv_SYSTEM.sys_get_ipmi_console()  # IPMIConsole object not the sol session object
con = console.get_console()                                      # Get the actual IPMI SOL Console
self.cv_IPMI.ipmi_host_login(con)                             # Login to the host on IPMI console
self.cv_IPMI.ipmi_host_set_unique_prompt()           # Set unique prompt
console.run_command(cmd, timeout)                      # Run the actual commands on host console

I will check with other testcases too, if anything brokes related to sol console.